firmware/system.c

```diff
@@ -20,30 +20,225 @@
  */
 
 #include "system.h"
+#include "settings.h"
+#include "ch.h"
+#include "hal.h"
+#include "logging.h"
 
-bool system_initialized = false;
-bool pid_request_active = false;
+#define _LOG_PFX "SYS:         "
 
+/* Flag to indicate if system is initialized
+ * and ready for normal operation */
+static bool system_initialized = false;
```

Comment:
Consider placing several of these globals into logical structs.  It makes code a bit smaller and is helpful when resetting state and generally keeping track of what goes where.